### PR TITLE
Add confirmation after clicking delete

### LINKED ISSUE #152: Add confirmation after clicking "Delete" button
## Expected Behavior
When a user wants to delete a device, they click the "Delete" button and the action is completed.  It would be preferable to have a confirmation that asks if they are sure that they want to delete the device.  It should also work well when viewed on a mobile device.

## Current Behavior
The device is immediately deleted when the user clicks the "Delete" button.

image

## Possible Solution
It would be acceptable to have a small modal/dialog box appear that says something like "Are you sure you want to delete [insert device name]?" with a "Yes" and a "No" button.  The default selected behavior should be the "No" action so a user doesn't excitedly delete a device.  This also means that pressing the "Enter" button on the keyboard will not have an effect.  It will also be necessary to add a Dusk test case or two to verify this behavior.

### PR DESCRIPTION
**Pull request by kailichiang**
Resolve #152

- Add confirmation javascript code
- Modify the test case for deletion
- Add a test case for canceling deletion

## Description
Add confirmation javascript code. This can be ubiquitous for any confirmation needs by adding class="confirmation".

## Motivation and Context

Confirmation before any un-redoable action is great.

[Add confirmation after clicking "Delete" button](https://github.com/dbudwin/RoboHome-Web/issues/152)

## How Has This Been Tested?

The change causes a js confirmation dialog before actual action. Add dialog check in test cases.

test by laravel dusk `php artisan dusk`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement/Security (non-breaking change which is not noticeable to end users)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I created a feature branch and did not open a pull request from my `master` branch.
- [x] My code follows the code style of this project.
- [ ] My change requires an update to the README and I have updated it accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] This is a complete change and doesn't leave the project in a bad state.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by kailichiang** [comment]
By my current implementation, it’s not possible to focus on cancel by default because it simply uses pure JavaScript confirm dialog. To achieve your possible solution, it  needs to create a custom modal. Surely it’s not a big problem. 

However, it’s obvious that you strongly want to prevent a user deleting a device by mistake. Therefore, I have 2 different solutions for your reference. 
1. Pop up a dialog and ask a user to input "the same device name." The device will be deleted if and only if the names are exactly the same. 
2. Provide an undo link followed by deletion message after deleted. This needs more efforts. You probably need to break down and create different issues, so people can get T-shirt by doing your project lol. 
    1. Change the migration to soft delete. It probably effects other functions. Fortunately you have test cases. 
    2. Provide a undo option route, like what delete does. 
    3. Permanent delete devices after a time by scheduling or any mechanism. 
    4. Add more test cases.
    5. Probably more. 

Just for your information.

**Comment by dbudwin** [comment]
While both of your proposed solutions would work, I think it may be a bit of over-engineering and a lot of extra work.

Hopefully this suggestion will be pretty straightforward.  How would you feel about using SweetAlert?  They have an example on their page that's pretty much what we're looking for and you may be able to simply copy and paste their example with some minor modifications.  It's the [last example in their "Getting Started" section](https://sweetalert.js.org/guides/#getting-started).  The code looks like this:

```
swal({
  title: "Are you sure?",
  text: "Once deleted, you will not be able to recover this imaginary file!",
  icon: "warning",
  buttons: true,
  dangerMode: true,
})
.then((willDelete) => {
  if (willDelete) {
    swal("Poof! Your imaginary file has been deleted!", {
      icon: "success",
    });
  } else {
    swal("Your imaginary file is safe!");
  }
});
```

If you like this route, I would suggest using `npm` with Webpack to install it (if you have experience with these tools) so it can easily be reused throughout this website.

**Comment by kailichiang** [comment]
Sweetalert is a great solution. I have pushed a new version by your suggestion. Update includes
1. Use sweetalert to show confirmation message.
2. Press enter to cancel delete.
3. Include device name in confirmation message.

In addition, the project doesn't really use Laravel mix, `npm` and `webpack`. That is, it doesn't use webpack to compile js and css files, and use compiled js and css files in blade templates. Therefore, I still use CDN to include sweetalert, as same as it uses bootstrap and jquery.

File: resources/views/devices.blade.php [hunk 1]
```diff
@@ -34,6 +35,23 @@ function controlDevice(action, publicDeviceId) {
                     url: "/devices/" + action + "/" + publicDeviceId
                 });
             }
+
+            function confirmBeforeDelete(that) {
```
Comment (dbudwin) [inline resources/views/devices.blade.php]: Great function name! 🎉

File: resources/views/devices.blade.php [hunk 2]
```diff
@@ -34,6 +35,23 @@ function controlDevice(action, publicDeviceId) {
...
+                let deviceName = $(that).data("device-name");
+                let deleteLink = $(that).data("delete-link");
+
+                swal({
+                    title: "Are you sure you want to delete [" + deviceName + "]?",
```
Comment (dbudwin) [inline resources/views/devices.blade.php]: I pulled down your pull request and started playing with it.  It works great!  I think I may want to tweak my idea a bit though to make it display better now that I can actually see this working...how about modifying this alert to be the following?

```
title: deviceName,
text: "Are you sure you want to delete this device?",
```

I think this makes it display nicely on both desktop and mobile.

screen shot 2018-10-15 at 18 14 42

**Comment by dbudwin** [comment]
Sigh, I accidentally merged this when messing with another branch locally.  Even if I can't fix it, [I see that you have already been given credit for this work on Hacktoberfest](https://hacktoberfest.digitalocean.com/stats/kailichiang) so don't be concerned about that.

**Comment by kailichiang** [comment]
Oops.... Is there anything I can do for you?

**Pull request merged**